DeviantArt/presence.ts

```diff
@@ -0,0 +1,565 @@
+var presence = new Presence({
+  clientId: "664057766809436161"
+});
+
+var currentURL = new URL(document.location.href),
+  currentPath = currentURL.pathname.slice(1).split("/"),
+  browsingStamp = Math.floor(Date.now() / 1000),
+  presenceData: presenceData = {
+    details: "Viewing an unsupported page",
+    largeImageKey: "lg",
+    startTimestamp: browsingStamp
+  },
+  updateCallback = {
+    _function: null as Function,
+    get function(): Function {
+      return this._function;
+    },
+    set function(parameter) {
+      this._function = parameter;
+    },
+    get present(): boolean {
+      return this._function !== null;
+    }
+  };
+
+/**
+ * Initialize/reset presenceData.
+ */
+function resetData(): void {
+  currentURL = new URL(document.location.href);
+  currentPath = currentURL.pathname.slice(1).split("/");
+  presenceData = {
+    details: "Viewing an unsupported page" as string,
+    state: undefined as string,
+    largeImageKey: "lg" as string,
+    startTimestamp: browsingStamp as number,
+    endTimestamp: undefined as number

```

Comment:
You still haven't fixed this, you know you can't just set null or undefined to a variable that's already in that state.

Reply:
That's the reset function.

Reply:
Instead of resetting just don't send them..
`Delete presenceData.state`

Reply:
Well, defining something as undefined is the same as not defining something.  
Still, guys, you guys keep resisting. I thought we're done about #1267. Fine.  
@doomlerd, hope you read my notes.